src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -512,95 +568,60 @@ static void freeMemoryRestricted(MemoryAddress addr) {
-         * from this va list will result in an exception.
-         *
-         * @see #isAlive()
-         */
-        void close();
+        ResourceScope scope();
 
         /**
          * Copies this C {@code va_list} at its current position. Copying is useful to traverse the va list's elements
          * starting from the current position, without affecting the state of the original va list, essentially
          * allowing the elements to be traversed multiple times.
          * <p>
-         * If this method needs to allocate native memory for the copy, it will use
-         * {@link MemorySegment#allocateNative(long, long)} to do so. {@link #close()} will have to be called on the
-         * returned va list instance to release the allocated memory.
+         * Any native resource required by the execution of this method will be allocated in the resource scope
+         * associated with this instance (see {@link #scope()}).
          * <p>
          * This method only copies the va list cursor itself and not the memory that may be attached to the
          * va list which holds its elements. That means that if this va list was created with the
          * {@link #make(Consumer)} method, closing this va list will also release the native memory that holds its
          * elements, making the copy unusable.
          *
          * @return a copy of this C {@code va_list}.
-         * @throws IllegalStateException if the C {@code va_list} associated with this instance is no longer valid
-         * (see {@link #close()}).
+         * @throws IllegalStateException if the resource scope associated with this instance has been closed
+         * (see {@link #scope()}).
          */
         VaList copy();
 
-        /**
-         * Copies this C {@code va_list} at its current position. Copying is useful to traverse the va list's elements
-         * starting from the current position, without affecting the state of the original va list, essentially
-         * allowing the elements to be traversed multiple times.
-         * <p>
-         * If this method needs to allocate native memory for the copy, it will use
-         * the given {@code NativeScope} to do so.
-         * <p>
-         * This method only copies the va list cursor itself and not the memory that may be attached to the
-         * va list which holds its elements. That means that if this va list was created with the
-         * {@link #make(Consumer)} method, closing this va list will also release the native memory that holds its
-         * elements, making the copy unusable.
-         *
-         * @param scope the scope to allocate the copy in
-         * @return a copy of this C {@code va_list}.
-         * @throws IllegalStateException if the C {@code va_list} associated with this instance is no longer valid
-         * (see {@link #close()}).
-         */
-        VaList copy(NativeScope scope);
```

Comment:
I double checked - I think the problem is that some implementations need to allocate, some don't. On SysV and Aarch, it seems to make sense to pass a scope, since you can actually create a new VaList, with a completely different scope. But on Windows? It seems like we're just creating another instance, backed by the same segment - so it's not clear how the scope would be used here. In other words, on Windows, the specified scope will say nothing about the lifecycle of the copy.